compiler/src/main/scala/org/scalajs/core/compiler/GenJSCode.scala

```diff
@@ -2540,15 +2540,25 @@ abstract class GenJSCode[G <: Global with Singleton](val global: G)
 
       if (enclosingLabelDefParams.contains(sym)) {
         genEnclosingLabelApply(tree)
-      } else if (countsOfReturnsToMatchEnd.contains(sym)) {
-        /* Jump the to the end-label of a pattern match
-         * Such labels have exactly one argument, which is the result of
-         * the pattern match (of type BoxedUnit if the match is in statement
-         * position). We simply `return` the argument as the result of the
-         * labeled block surrounding the match.
+      } else if (countsOfReturnsToPatmatLabel.contains(sym)) {
+        /* Jump the to a label of a pattern match.
+         *
+         * Such labels are not enclosing. Instead, they are forward jumps to a
+         * following case, or to the match-end. For those labels, we generate a
+         * js.Return and keep track of how many such returns we generate, so
+         * that the enclosing `genTranslatedMatch` can optimize away the
+         * labeled blocks in some cases, notably when they are not used at all.
+         *
+         * Next-case labels have no argument, while match-end labels have
+         * exactly one argument, which is the result of the pattern match (of
+         * type BoxedUnit if the match is in statement position). In the latter
```

Comment:
Just to check my understanding: We rely on the arguments to determine if this is a 0 or 1 argument label?